client/js/table-view.js

```diff
@@ -1,88 +1,214 @@
+                const position = { col: col - 1, row: row };
+                const value = parseInt(this.model.getValue(position))
+                if ((typeof value === 'number') && !isNaN(value)) {
+                    sumData[col] = sumData[col] + value;
+                }
+            }
+        }
+        removeChildren(this.sumBar);
+        sumData.map(colSum => createTD(colSum)).forEach(cs => this.sumBar.appendChild(cs));
+    }
+
+    // returns if the given location (in col, row) matches actual location  
     isCurrentCell(col, row) {
         return this.currentCellLocation.col === col &&
             this.currentCellLocation.row === row;
     }
+
+    // returns if the given column location (in col) matches actual column location
+    isCurrentCol(col) {
+        return this.currentCellLocation.col === col &&
+            this.model.highlight.col === true;
+    }
+
+    // returns if the given row location (in row) matches actual row location
+    isCurrentRow(row) {
+        return this.currentCellLocation.row === row &&
+            this.model.highlight.row === true;
+    }
+
+    //renders table body, updates class of table cells created, and creates the first table row
     renderTableBody() {
         const fragment = document.createDocumentFragment();
         for (let row = 0; row < this.model.numRows; row++) {
+            const th = createTH(row + 1);
+            th.setAttribute('class', 'headerColumn');
             const tr = createTR();
+            tr.appendChild(th);
             for (let col = 0; col < this.model.numCols; col++) {
                 const position = { col: col, row: row }
                 const value = this.model.getValue(position);
                 const td = createTD(value);
 
                 if (this.isCurrentCell(col, row)) {
                     td.className = 'current-cell';
+                } else if (this.isCurrentCol(col + 1)) {
+                    td.className = 'current-column';
+                } else if (this.isCurrentRow(row)) {
+                    td.className = 'current-row';
                 }
 
-                tr.appendChild(td);
+                tr.appendChild(td)
+
             }
             fragment.appendChild(tr);
         }
         removeChildren(this.sheetBodyEl);
         this.sheetBodyEl.appendChild(fragment);
     }
+
+    //Attaches event handlers to the specified DOM elements
     attachEventHandlers() {
         this.sheetBodyEl.addEventListener('click', this.handleSheetClick.bind(this));
         this.formulaBarEl.addEventListener('keyup', this.handleFormulaBarChange.bind(this));
+        this.columnButton.addEventListener('click', this.addColumn.bind(this));
+        this.rowButton.addEventListener('click', this.addRow.bind(this));
+        this.headerRowEl.addEventListener('click', this.handleColumnClick.bind(this));
     }
 
+    //Adds column when called. Called by click columnButton.
+    addColumn() {
```

Comment:
This ended up being a large function with lots of nesting. Could you make it cleaner by using a helper function?